src/GitHub.InlineReviews/PullRequestStatusBarPackage.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Threading;
+using System.Runtime.InteropServices;
+using GitHub.Helpers;
+using GitHub.Services;
+using GitHub.VisualStudio;
+using GitHub.InlineReviews.Services;
+using Microsoft.VisualStudio.Shell;
+using Task = System.Threading.Tasks.Task;
+
+namespace GitHub.InlineReviews
+{
+    [Guid(Guids.PullRequestStatusPackageId)]
+    [PackageRegistration(UseManagedResourcesOnly = true, AllowsBackgroundLoading = true)]
+    [ProvideAutoLoad(Guids.GitSccProviderId, PackageAutoLoadFlags.BackgroundLoad)]
+    public class PullRequestStatusBarPackage : AsyncPackage
+    {
+        /// <summary>
+        /// Initialize the PR status UI on Visual Studio's status bar.
+        /// </summary>
+        protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
+        {
+            var usageTracker = (IUsageTracker)await GetServiceAsync(typeof(IUsageTracker));
+            var serviceProvider = (IGitHubServiceProvider)await GetServiceAsync(typeof(IGitHubServiceProvider));
+
+            await ThreadingHelper.SwitchToMainThreadAsync();
```

Comment:
There seems to be an issue where this await doesn't return until a solution is loaded. This means that when Visual Studio is opened without a solution, the PR list might be visible but the current PR doesn't appear on the status bar.

This isn't necessarily a problem, but I'd like to understand what's going on. I wonder if this will impact any of our other async loading packages? We wouldn't notice the issue on packages that use `[ProvideAutoLoad(UIContextGuids80.SolutionExists)]` (because they only load when a solution exists).

Reply:
Yeah, this is yet another reason why we shouldn't be switching to the main thread for initialization whenever we can avoid it. If all we need are non-MEF services, we shouldn't need to switch to the main thread for any of this.